run.sh

```diff
@@ -1,15 +1,71 @@
 #!/bin/bash
 set -e
 set -x
-export DEVPI_SERVERDIR=/mnt
-export DEVPI_CLIENTDIR=/tmp/devpi-client
-[[ -f $DEVPI_SERVERDIR/.serverversion ]] || initialize=yes
-
-devpi-server --start --host 0.0.0.0 --port 3141
-if [[ $initialize = yes ]]; then
-  devpi use http://localhost:3141
-  devpi login root --password=''
-  devpi user -m root password="${DEVPI_PASSWORD}"
+
+
+#===========================
+# Default environment if none 
+# is provided
+#===========================
+DEFAULT_DEVPI_SERVERDIR=/devpidata
+DEFAULT_NGINX_IP=0.0.0.0
+DEFAULT_NGINX_PORT=3141
+DEFAULT_DEVPI_SERVERIP=127.0.0.1
+DEFAULT_DEVPI_SERVERPORT=3142
+DEFAULT_DEVPI_CLIENTDIR=/tmp/devpi-client
+DEFAULT_DEVPI_CLIENTNAME=root
+DEFAULT_DEVPI_CLIENTPWD=
+DEFAULT_DEVPI_LDAPCONFILE=
+
+# http://stackoverflow.com/questions/228544/how-to-tell-if-a-string-is-not-defined-in-a-bash-shell-script
+[[ -z ${DEVPI_SERVERDIR+xxx} ]]   && export DEVPI_SERVERDIR=$DEFAULT_DEVPI_SERVERDIR
+[[ -z ${NGINX_IP+xxx} ]]          && export NGINX_IP=$DEFAULT_NGINX_IP
+[[ -z ${NGINX_PORT+xxx} ]]        && export NGINX_PORT=$DEFAULT_NGINX_PORT
+[[ -z ${DEVPI_SERVERIP+xxx} ]]    && export DEVPI_SERVERIP=$DEFAULT_DEVPI_SERVERIP
+[[ -z ${DEVPI_SERVERPORT+xxx} ]]  && export DEVPI_SERVERPORT=$DEFAULT_DEVPI_SERVERPORT
+[[ -z ${DEVPI_CLIENTDIR+xxx} ]]   && export DEVPI_CLIENTDIR=$DEFAULT_DEVPI_CLIENTDIR
+[[ -z ${DEVPI_CLIENTNAME+xxx} ]]  && export DEVPI_CLIENTNAME=$DEFAULT_DEVPI_CLIENTNAME
+[[ -z ${DEVPI_CLIENTPWD+xxx} ]]   && export DEVPI_CLIENTPWD=$DEFAULT_DEVPI_CLIENTPWD
+[[ -z ${DEVPI_LDAPCONFILE+xxx} ]] && export DEVPI_LDAPCONFILE=$DEFAULT_DEVPI_LDAPCONFILE
+
+#===========================
+# devpi configuration
+# TODO add ldap stuff
+#===========================
+
+# Start the server
+devpi-server --start --host ${DEVPI_SERVERIP} --port ${DEVPI_SERVERPORT}
+
+# Configure a default client if necessary
+if [[ ! -f ${DEVPI_SERVERDIR}/.serverversion ]]; then
+  devpi use http://${DEVPI_SERVERIP}:${DEVPI_SERVERPORT}
+  devpi login ${DEVPI_CLIENTNAME} --password="${DEVPI_CLIENTPWD}"
+  devpi user -m root password="${DEVPI_CLIENTPWD}"
```

Comment:
Although DEVPI_CLIENTPWD seems more accurate, the old name DEVPI_PASSWORD is still mentioned in README and we should retain backwards compatibility.